Tasks/replace.js

```diff
@@ -1,22 +1,34 @@
+'use strict';
 // Replace substring with newstr
 
-Replace = (str, substr, newstr) => {
+const replace = (str, substr, newstr) => {
+  if (
+    typeof str !== 'string' ||
+    typeof substr !== 'string' ||
+    typeof newstr !== 'string'
+  ) {
+    throw new Error('str, substr or newstr are not a String');
+  }
+
   if (substr === '') {
     return str;
-  } else {
-    src = str;
-    res = '';
-    do {
-      const _index = src.indexOf(substr);
-      if (_index === -1) {
-        return res + src;
-      } else {
-        const start = src.substring(0, _index);
-        src = src.substring(_index + substr.length, src.length);
-        res += start + newstr;
-      }
-    } while (true);
   }
+
+  let result = str;
+
+  do {
+    const startIndex = result.indexOf(substr);
+
+    if (startIndex === -1) {
+      return result;
+    }
+
+    const endIndex = startIndex + substr.length;
+    const startString = result.substring(0, startIndex);
+    const endString = endIndex > result.length ?
+      '' : result.substring(endIndex);
+    result = startString + newstr + endString;
+  } while (true);
 };
```

Comment:
Not optimal but here is `replaceAll` implementation that is even better, we need just single replace